src/start/IosCompliantAppleAuthButton/index.js

```diff
@@ -15,7 +16,13 @@ type SelectorProps = $ReadOnly<{|
 |}>;
 
 type Props = $ReadOnly<{|
-  style?: ViewStyleProp,
+  /* eslint-disable flowtype/generic-spacing */
+  style?: SubsetProperties<
+    ViewStyle,
+    {|
+      marginTop?: mixed,
+    |},
+  >,
```

Comment:
Partly in the spirit of 8b30fa9d6, but also because this helps us
    avoid a mystifying error reported by Flow when we take the upgrade
    to Flow v0.122.0, with the React Native v0.63 upgrade (#4245).
    
    I was seeing the error on the use of `ZulipButton` in `AuthScreen`;
    it didn't like what I was passing for the `style` prop. When I
    temporarily commented out the `style` prop passed to
    `IosCompliantAppleAuthButton`, the error disappeared. It doesn't
    make a lot of sense to me, but it sounds like
    `IosCompliantAppleAuthButton`'s type for its `style` prop was
    influencing what type `styles.halfMarginTop` was inferred to have,
    when it was passed to `ZulipButton`.

Huh wacky. That seems like a Flow bug... quite likely reporting a real type mismatch, but a bug in at least where it places the error. I'd be curious to take a look at the error -- want to paste it somewhere, perhaps in chat?

In any case this seems like a fine tightening of the type. Perhaps add a comment pointing at ZulipButton's props, where a comment discusses this idea.

Reply:
Sure, I've just posted the error(s!) in chat, [here](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Flow.20errors.20in.20.23M4318/near/1075420).